core/server/services/rss/generate-feed.js

```diff
@@ -87,16 +87,17 @@ generateFeed = function generateFeed(baseUrl, data) {
             }
         });
 
-    data.posts.forEach(function forEach(post) {
-        var item = generateItem(post, siteUrl, data.secure);
-
-        filters.doFilter('rss.item', item, post).then(function then(item) {
-            feed.item(item);
+    return data.posts.reduce(function (feedPromise, post) {
+        return feedPromise.then(function () {
+            var item = generateItem(post, siteUrl, data.secure);
+            return filters.doFilter('rss.item', item, post).then(function (item) {
+                return feed.item(item);
+            });
+        });
+    }, Promise.resolve()).then(function () {
```

Comment:
We are using Bluebird, not the native Promise. We overriding the globals, but we are still requiring bluebird on top for now till we find a better way. My IDE for example marks this as "missing require".

Reply:
I know you do ) Are you sure? Because Bluebird provides the `resolve` function too: http://bluebirdjs.com/docs/api/promise.resolve.html

Reply:
I don't understand - we are using Bluebird.

Reply:
Ok, I've explicitly required it.